torch/testing/_internal/common_methods_invocations.py

```diff
@@ -10549,11 +10549,6 @@ def ref_pairwise_distance(input1, input2):
                     supports_autograd=False,
                     # FIXME: heaviside does not accept scalar inputs
                     skips=(
-                        # NumPy's heaviside promotes bool to float16
-                        DecorateInfo(unittest.expectedFailure,
-                                     'TestBinaryUfuncs',
-                                     'test_reference_numerics_heavisidel',
-                                     dtypes=(torch.bool,)),
```

Comment:
The test name is malformed. It should be `test_reference_numerics` rather than `test_reference_numeric_heavisidel`. As a consequence, this test was never skipped and passed "unexpectedly" with the current behavior of `assertEqual`. This showed up, because at the time the comparison logic did not handle `torch.tensor(True)` vs `np.float16(1.0)` well. This is now fixed.